pkg/settings/setting_test.go

```diff
@@ -76,3 +79,42 @@ func TestSystemFeatureChartRefreshSecondsDefault(t *testing.T) {
 	}
 
 }
+
+func TestGetMachineProvisionImagePullPolicy(t *testing.T) {
+	defalutLogger := logrus.StandardLogger().Out
+	logrus.SetOutput(io.Discard)
+	defer func() {
+		logrus.SetOutput(defalutLogger)
+	}()
+
+	testCases := []struct {
+		settingValue string
```

Comment:
For tests, I use the `input/expected` format for test cases. This is also throughout the Rancher code. Here's a good example https://github.com/rancher/rancher/blob/59b00c6f9a0a7c598e24c491c045a0fc0586ba6b/pkg/catalogv2/system/system_test.go#L34-L50.